.svglintrc.js

```diff
@@ -163,6 +163,65 @@ module.exports = {
               }
             }
           },
+          function(reporter, $, ast) {
+            reporter.name = "icon-path";
+
+            const iconPath = $.find("path").attr("d");
+            if (!updateIgnoreFile && isIgnored(reporter.name, iconPath)) {
+              return;
+            }
+
+            const { segments } = parsePath(iconPath);
+
+            const lowerCommand = ['m', 'l'];
+            const lowerDirectionCommand = ['h', 'v'];
+            const upperCommand = ['M', 'L'];
+            const upperHorDirectionCommand = 'H';
+            const upperVerDirectionCommand = 'V';
+            const upperDirectionCommand = [upperHorDirectionCommand, upperVerDirectionCommand];
+            const commands = [...lowerCommand, ...lowerDirectionCommand, ...upperCommand, ...upperDirectionCommand];
+            const getInvalidSegments = ([command, coord1, coord2, ...rest], index) => {
+              if (commands.includes(command)) {
+                // Relative directions (h or v) having a length of 0
+                if (lowerDirectionCommand.includes(command) && coord1 === 0) {
+                  return true;
+                }
+                // Relative movement (m or l) having a distance of 0
+                if (lowerCommand.includes(command) && coord1 === 0 && coord2 === 0) {
+                  return true;
+                }
+                if (index > 0) {
+                  const [prevCoord2, prevCoord1, ...rest] = [...segments[index - 1]].reverse();
+                  // Absolute horizontal direction (H) having the same x coordinate as the previous segment
+                  if (upperHorDirectionCommand === command && coord1 === prevCoord1) {
```

Comment:
This does not work as expected right now. For example, it gave the following error for Zendesk: `Unexpected segment H12.915 in path.` that looks like 
![image](https://user-images.githubusercontent.com/12817388/101204985-151b1980-363b-11eb-9851-7277d5bfc0a4.png)

If I remove this segment from the path, the icon looks like
![image](https://user-images.githubusercontent.com/12817388/101205044-2a904380-363b-11eb-850a-d715a426ed95.png)

Reply:
Check the previous command, is `m` (relative): `m12.915 0H12.915`

Reply:
Yes, not that easy to have a good inefficiency checker... I would need to keep the current computed absolute coordinate when looping over segments... I think it will become an overcomplicated rule...

`l10,0 l10,10 l10,0 H30` is inefficient. But to know it is, I would need to compute the current absolute coordinate after each segments:
1. [l, 10, 0]: (10, 0)
2. [l, 10, 10]: (20, 0)
3. [l, 10, 0]: (30, 0)
4. [H, 30]: invalid since the previous coordinate was (30, 0)

Can look easy while taking lineTo or moveTo, but computing the final coordinate when using arc or curve and when mixing absolute and relative command, it becomes complicated.
Maybe i'm also missing something 😛

Reply:
I think that you are well on your way. I ran into a similar problem developing `svg-path-bbox`. The solution would be to absolutify the points in another array to compare with it. You can use the `abs()` function that `svgpath` provides, something like:

```javascript
svgPath(d).abs().unarc().unshort().iterate((seg, dc, x, y) => { ... });
```

Reply:
Thank you very much for your help on this!
`svgPath(d).abs().unarc().unshort()` is working well to get only absolute coordinate, but the number of segments is not always the same as the parsePath results. In this case, how can I compare the 2 arrays since item index wont necessarily be the same between those arrays?

Reply:
If I don't get lost, you "just" need to do for `l`, `L`, `m`, `M`, `v`, `V`, `h` and `H` commands right? For `Q` and `C` segments returned by `abs().unarc().unshort()`, the last coordinate will be those numbers in the indexes 5 and 6 for `C` and 3 and 4 for `Q`.

Reply:
:thinking: You have tried with `svgPath(d).abs().unshort().iterate((seg, dc, x, y) => { ... }` (without `unarc()`)?

Reply:
Ha, I should have think of that!
Now, both returns the same number of segments.
Will try to see if I'm able to use this!

Reply:
By the way, keep in mind that whitout unarc() you must handle also `S` and `T` commands.

Reply:
Looks better now! Thanks again!
There is another edge case 😅 when I compare with the previous path. If the previous path is an absolute direction command (H or V), I only have a part of the coordinate (X or Y). 

So if I have `['L', 0, 24]` and the previous command was a direction one, I'm not able to tell if it is inefficient. In those cases, I would need to check the previous of the previous until I find the missing coordinate.

ex.
- `['L', 12, 24]` (12, 24)
- `['H', 12]` (12, ?)
- `['V', 24]` (12, 24)

Reply:
Wow that's interesting! So you must iterate backwards more for `H` and `V` commands until `H` and `V` (both) have been found or another command, not?. Don't worry if the code is too long, we can always separate it into another package. I'm sure that this implementation could be very reusable.

Reply:
Exact! It could be any command that contains the missing coordinate. So if the last one is `H24` (`x` = 24), I need to know what was the previous value for `y`. So I need to iterate back to find a `V` or any other command that contains a coordinate for `y`.
I have something in mind. Stopping for today (friday 5pm here, time for a beer 😛)

Reply:
More tests are needed, but I think I have something working!

Reply:
I will try to review this as soon as possible.